### PR TITLE
janet: 1.31 -> 1.32.1

### PR DESCRIPTION
**Pull request by cafkafk**
## Description of changes
Changelog: https://github.com/janet-lang/janet/releases/tag/v1.32.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lucasew** [comment]
rename the PR and the commit to `janet: 1.31 -> 1.32.0`

**Comment by cafkafk** [comment]
> rename the PR and the commit to `janet: 1.31 -> 1.32.0`

whoops... just pushed fix

**Comment by lucasew** [comment]
BTW r-ryantm can take over next time with this:

```patch
diff --git a/pkgs/development/interpreters/janet/default.nix b/pkgs/development/interpreters/janet/default.nix
index 3c44239dc2fa..1de6b76e063f 100644
--- a/pkgs/development/interpreters/janet/default.nix
+++ b/pkgs/development/interpreters/janet/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, meson, ninja }:
+{ lib, stdenv, fetchFromGitHub, meson, ninja, nix-update-script }:
 
 stdenv.mkDerivation rec {
   pname = "janet";
@@ -29,6 +29,8 @@ stdenv.mkDerivation rec {
     $out/bin/janet -e '(+ 1 2 3)'
   '';
 
+  passthru.updateScript = nix-update-script {};
+
   meta = with lib; {
     description = "Janet programming language";
     homepage = "https://janet-lang.org/";
```

**Comment by cafkafk** [comment]
Result of `nixpkgs-review pr 261147` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>janet</li>
    <li>jpm</li>
  </ul>
</details>

**Comment by cafkafk** [comment]
> BTW r-ryantm can take over next time with this:

Ohh, cool, applied patch in https://github.com/NixOS/nixpkgs/pull/261147/commits/40ee8349089b9a4cc2f4be8a3be79038c6664203

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 261147` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>janet</li>
    <li>jpm</li>
  </ul>
</details>

**Review by lucasew — APPROVED** [review]

**Comment by cafkafk** [comment]
I added some very simple `passthru.tests` which might help speed up the review process for the ryanbot in the future

File: pkgs/development/interpreters/janet/default.nix [hunk 1]
```diff
@@ -29,11 +36,26 @@ stdenv.mkDerivation rec {
     $out/bin/janet -e '(+ 1 2 3)'
   '';
```
Comment (Artturin) [inline pkgs/development/interpreters/janet/default.nix:38]:
```suggestion
  # Tests require binding to local addresses for listening sockets.
  __darwinAllowLocalNetworking = true;
```

This should fix the darwin test failure

```
Starting suite /private/tmp/nix-build-janet-1.32.0.drv-0/source/build/../test/suite-ev.janet...
error: Socket is not connected
  in net/write [src/core/net.c] on line 893
  in <anonymous> [/private/tmp/nix-build-janet-1.32.0.drv-0/source/build/../test/suite-ev.janet] on line 200, column 7
  in test-echo [/private/tmp/nix-build-janet-1.32.0.drv-0/source/build/../test/suite-ev.janet] on line 199, column 5
  in _while [/private/tmp/nix-build-janet-1.32.0.drv-0/source/build/../test/suite-ev.janet] on line 204, column 3
  in _thunk [/private/tmp/nix-build-janet-1.32.0.drv-0/source/build/../test/suite-ev.janet] (tailcall) on line 183, column 1
```

**Comment by lucasew** [comment]
Result of `nixpkgs-review pr 261147` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>janet</li>
    <li>jpm</li>
  </ul>
</details>

File: pkgs/development/interpreters/janet/default.nix (hunk 1 above)
Comment (cafkafk) [reply pkgs/development/interpreters/janet/default.nix:38]: it didn't work it seems, not sure how to proceed

Comment (Artturin) [reply pkgs/development/interpreters/janet/default.nix:38]: Let's disable that test on darwin 

Removing this line https://github.com/janet-lang/janet/blob/cc5beda0d26c0b91fb28540291e950392cf59fc4/meson.build#L246 should work, There's no other mention of `suite-ev.janet` in the repo.

**Review by lucasew — APPROVED** [review]

File: pkgs/development/interpreters/janet/default.nix [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
+    hash = "sha256-24d9N59pTfQATWmAZN4dAFT8RTTlUlBPKokcQ/Fd2No=";
   };
 
   postPatch = ''
     substituteInPlace janet.1 \
       --replace /usr/local/ $out/
+  '' + lib.optionalString stdenv.isDarwin ''
+    # error: Socket is not connected
+    substituteInPlace meson.build \
+      --replace "'test/suite-ev.janet'," ""
```
Comment (Artturin) [inline pkgs/development/interpreters/janet/default.nix:27]: Converted the patch to this

**Pull request merged**